logos/logos-orchestrator/src/logos/capacity/capacity_planner.py

```
@@ -5463,17 +5496,51 @@ def _estimate_available_for_kv_mb(
+        # than stacking on top of it, so the figure that actually competes for
+        # VRAM is the weights-only footprint. Subtracting the baked-in KV is
+        # essential: without it, available-for-KV is under-counted by a whole
+        # kv_budget (e.g. a 35B lane reports ~0.5 GiB free when ~10 GiB is
+        # really available), every calibrated pair is rejected, and the planner
+        # falls back to a floor KV with the worker's full-context max_model_len
+        # — a split-brain launch vLLM rejects outright.
+        #
+        # Mind the units: kv_budget_mb / max_kv_cache_mb are PER-RANK budgets
+        # (same units as kv_cache_memory_bytes and the pair curve's kv_mb),
+        # whereas base_residency_mb is the TOTAL footprint across all TP GPUs.
+        # So the KV baked into base spans every rank — subtract budget × tp, not
```

Ruff (0.15.17)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)